core/src/serve_repair.rs

```diff
@@ -445,9 +464,103 @@ impl ServeRepair {
         }
     }
 
+    fn decode_requests(
+        thread_pool: &ThreadPool,
+        reqs_v: Vec<PacketBatch>,
+        epoch_staked_nodes: &Option<Arc<HashMap<Pubkey, u64>>>,
+        whitelist: &RwLockReadGuard<HashSet<Pubkey>>,
+        my_id: &Pubkey,
+        socket_addr_space: &SocketAddrSpace,
+        stats: &mut ServeRepairStats,
+        cluster_type: ClusterType,
+    ) -> Vec<RepairRequestWithMeta> {
+        thread_pool
+            .install(|| {
+                reqs_v
+                    .par_iter()
+                    .with_min_len(VERIFY_MIN_PACKETS_PER_THREAD)
```

Comment:
Per my observations, `.with_min_len` still adds overhead if total number of packets is smaller than this min-len.
Might be better not to use thread-pool at all in that case; something like:
https://github.com/solana-labs/solana/blob/f24e30e4f/ledger/src/blockstore.rs#L2803-L2824